processor/deltatocumulativeprocessor/README.md

```diff
@@ -29,4 +29,21 @@ processors:
         # upper limit of streams to track. new streams exceeding this limit
         # will be dropped
         [ max_streams: <int> | default = 0 (off) ]
+
 ```
+
+There is no further configuration required. All delta samples are converted to cumulative.
+
+## Troubleshooting
+
+The following metrics are recorded when [telemetry is
+enabled](https://opentelemetry.io/docs/collector/configuration/#telemetry):
+
+| Name                                     | Description                                                                           | Unit          |
+|------------------------------------------|---------------------------------------------------------------------------------------|---------------|
+| `deltatocumulative.streams.tracked`      | Number of streams currently tracked by the aggregation state                          | `{stream}`    |
```

Comment:
Please correct me if I'm missing something, I don't know this processor's code well. My assumption would be for any metrics emitted by a component to have them declared in `metadata.yaml` file and then use the generated code in `internal/metadata/generated_metrics.go` file to emit those. Many components do this already, see for example https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.96.0/receiver/httpcheckreceiver/metadata.yaml#L31-L56.

Is there a reason for this component to not use this way to emit metrics? Perhaps I'm misunderstanding something.

Reply:
oh wow, I didn't know this even existed!!
guess I have some refactoring to do haha

Reply:
@astencel-sumo looking more into this, I'm not sure this is what we want here.

First of all, this is a processor, not a receiver. It does not exactly create new metrics, but transform the ones it receives.
While doing that, certain statistics about its behavior can be observed, but those are mostly useful for troubleshooting the collector. I'd want them to end up at https://opentelemetry.io/docs/collector/configuration/#telemetry

What you pointed me at would require sending those along the processed metrics to the output metrics consumer

Reply:
Oh, that's right, I totally mistook the two types of metrics :facepalm: 

You are right. A component should only define in `metadata.yaml` file the metrics that it intends to emit to the telemetry pipeline. The "own metrics" that the component emits as the collector's telemetry do not go into `metadata.yaml`. Thanks for correcting me!